apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -0,0 +1,308 @@
+}
+
+type QueueingStrategy string
+
+const (
+	// StrictFIFO means that workloads are ordered strictly by creation time.
+	// Older workloads that can't be admitted will block admitting newer
+	// workloads even if they fit available quota.
+	StrictFIFO QueueingStrategy = "StrictFIFO"
+
+	// BestEffortFIFO means that workloads are ordered by creation time,
+	// however older workloads that can't be admitted will not block
+	// admitting newer workloads that fit existing quota.
+	BestEffortFIFO QueueingStrategy = "BestEffortFIFO"
+)
+
+type ResourceGroup struct {
+	// resources is the list of resources covered by the flavors in this group.
+	// Examples: cpu, memory, vendor.com/gpu.
+	// The list cannot be empty and it can contain up to 16 resources.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	Resources []corev1.ResourceName `json:"resources"`
+
+	// flavors is the list of flavors that provide the resources of this group.
+	// Typically, different flavors represent different hardware models
+	// (e.g., gpu models, cpu architectures) or pricing models (on-demand vs spot
+	// cpus).
+	// Each flavor MUST list all the resources listed for this group in the same
+	// order as the .resources field.
+	// The list cannot be empty and it can contain up to 16 flavors.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	Flavors []FlavorQuotas `json:"flavors"`
+}
+
+type FlavorQuotas struct {
+	// name of this flavor. The name should match the .metadata.name of a
+	// ResourceFlavor. If a matching ResourceFlavor does not exist, the
+	// ClusterQueue will have an Active condition set to False.
+	Name ResourceFlavorReference `json:"name"`
+
+	// resources is the list of quotas for this flavor per resource.
+	// There could be up to 16 resources.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=16
+	Resources []ResourceQuota `json:"resources"`
+}
+
+type ResourceQuota struct {
+	// name of this resource.
+	Name corev1.ResourceName `json:"name"`
+
+	// nominalQuota is the quantity of this resource that is available for
+	// Workloads admitted by this ClusterQueue at a point in time.
+	// The quota must non-negative.
+	// nominalQuota should represent the resources in the cluster available for
+	// running jobs (after discounting resources consumed by system components
+	// and pods not managed by kueue). In an autoscaled cluster, nominalQuota
+	// should account for resources that can be provided by a component such as
+	// Kubernetes cluster-autoscaler.
+	//
+	// If the ClusterQueue belongs to a cohort, the sum of the quotas for each
+	// (flavor, resource) combination defines the maximum quantity that can be
+	// allocated by a ClusterQueue in the cohort.
+	NominalQuota resource.Quantity `json:"nominalQuota"`
+
+	// borrowingLimit is the maximum amount of quota for the [flavor, resource]
```

Comment:
Maybe describe that `MaximumQuota = NominalQuota + BorrowingLimit` is more intuitive.

Reply:
See update.